azure/scope/strategies/machinepool_deployments/machinepool_deployment_strategy.go

```diff
@@ -233,12 +234,17 @@ func getFailedMachines(machinesByProviderID map[string]infrav1exp.AzureMachinePo
 	return machines
 }
 
+// getDeletingMachines is responsible for identifying machines whose VMs are in an active state of deletion
+// but whose corresponding AzureMachinePoolMachine resource has not yet been marked for deletion.
 func getDeletingMachines(machinesByProviderID map[string]infrav1exp.AzureMachinePoolMachine) []infrav1exp.AzureMachinePoolMachine {
 	var machines []infrav1exp.AzureMachinePoolMachine
 	for _, v := range machinesByProviderID {
-		// ready status, with provisioning state Succeeded, and not marked for delete
+		// provisioning state is Deleting
 		if v.Status.ProvisioningState != nil && *v.Status.ProvisioningState == infrav1.Deleting {
```

Comment:
I think this line should be deleted.

Reply:
Which line?

Reply:
https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/2085#discussion_r807316816

Consider combining the if statement.